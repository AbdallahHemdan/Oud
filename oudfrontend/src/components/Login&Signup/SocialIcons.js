import React, {Component} from 'react';
import Facebook from './signin/Facebook';
import axios from 'axios';
import '../../components/Login&Signup/signup/signup.css';

/**the social button section
 * @author abdallah abu sedo
 */
class SocialIcons extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  /**
   * the request sender function
   * @function
   * @returns {void}
   */
  handelClickFace = () => {
    let toSent = {
      access_token: this.state.access_token,
    };
    axios
      .post('https://oud-zerobase.me/api/v1/auth/facebook', toSent)
      .then((response) => {
        console.log(response);

        window.location = '/';
      })
      .catch((error) => {
        console.log(error.response);
      });
  };
  /**
   * render and call the buttons
   * @function
   * @returns {void}
   */
  render() {
    return (
      <React.Fragment>
        <div className="SocialIconsPage container">
          <h6 className="hint-text hint" data-testid="SocialText">
            Sign Up with your social media account
          </h6>
          <button className="google_button" href="#">
            <img
              data-testid="GoogleButtonImage"
              alt=""
              src="http://pngimg.com/uploads/google/google_PNG19630.png"
              className="googlepic"
            ></img>
            continue with Google
          </button>
          <Facebook
            data-testid="FacebookButton"
            onClick={this.handelClickFace}
          />

          <section className="or-seperator OR">
            <i>OR</i>
            <h6 className="hint-text hint">Sign up with your email address</h6>
          </section>
        </div>
      </React.Fragment>
    );
  }
}

export default SocialIcons;
