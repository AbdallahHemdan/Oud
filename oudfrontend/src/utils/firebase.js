import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
