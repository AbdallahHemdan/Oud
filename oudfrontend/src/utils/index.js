import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
