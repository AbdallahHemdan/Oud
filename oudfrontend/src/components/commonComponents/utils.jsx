import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}