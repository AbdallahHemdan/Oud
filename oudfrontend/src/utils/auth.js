import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
