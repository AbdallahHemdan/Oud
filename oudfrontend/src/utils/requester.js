import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
