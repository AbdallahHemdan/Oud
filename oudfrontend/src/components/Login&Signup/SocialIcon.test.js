import React from 'react';
import {shallow, mount} from 'enzyme';
import {findByTestAttr} from './../../utils/index';
import SocialIcons from './SocialIcons';

const setup = (props = {}) => {
  const component = shallow(<SocialIcons {...props} />);
  return component;
};
describe('test the SocialIcons', () => {
  let component;
  beforeEach(() => {
    const props = {};
    component = setup(props);
  });
  it('Render the SocialText text box ', () => {
    const wrapper = findByTestAttr(component, 'SocialText');
    expect(wrapper.length).toBe(1);
  });
  it('Render the GoogleButtonImage text box ', () => {
    const wrapper = findByTestAttr(component, 'GoogleButtonImage');
    expect(wrapper.length).toBe(1);
  });
  it('Render the FacebookButton text box ', () => {
    const wrapper = findByTestAttr(component, 'FacebookButton');
    expect(wrapper.length).toBe(1);
  });
  describe('Snapshot test', () => {
    it('renders component correctly', () => {
      const wrapper = shallow(<SocialIcons />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
