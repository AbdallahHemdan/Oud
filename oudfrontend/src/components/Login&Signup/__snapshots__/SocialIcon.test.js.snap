// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the SocialIcons Snapshot test renders component correctly 1`] = `ShallowWrapper {}`;
